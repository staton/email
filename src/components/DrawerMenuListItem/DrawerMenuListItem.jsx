import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

const propTypes = {
    badge: PropTypes.number,
    icon: PropTypes.element,
    onClick: PropTypes.func,
    primaryContent: PropTypes.any.isRequired,
    secondaryContent: PropTypes.any,
    title: PropTypes.string
};

const defaultProps = {
    badge: null,
    icon: null,
    onClick: () => {},
    secondaryContent: null,
    title: ''
};

export class DrawerMenuListItem extends Component {

	render() {
		return (
            <li 
                className="DrawerMenuListItem"
                title={this.props.title}>
                <div className="DrawerMenuListItem__left-container">
                    <div className="DrawerMenuListItem__icon">
                    {this.props.icon}
                    </div>
                    <div className="DrawerMenuListItem__content">
                        <div className="DrawerMenuListItem__primary-content">
                        {this.props.primaryContent}
                        </div>
                        {
                            (this.props.secondaryContent)
                            ?   <div className="DrawerMenuListItem__secondary-content">
                                {this.props.secondaryContent}
                                </div>
                            :   null
                        }
                    </div>
                </div>
                {
                    (this.props.badge)
                    ?   <div className="DrawerMenuListItem__badge">
                            <span>{this.props.badge}</span>
                        </div>
                    :   null
                }
            </li>
		);
    }

}

function mapStateToProps(store, ownProps) {
    return {
    };
}

function mapDispatchToProps(dispatch) {
    return bindActionCreators({
    },
    dispatch);
}

DrawerMenuListItem.propTypes = propTypes;
DrawerMenuListItem.defaultProps = defaultProps;

export default connect(mapStateToProps, mapDispatchToProps)(DrawerMenuListItem);