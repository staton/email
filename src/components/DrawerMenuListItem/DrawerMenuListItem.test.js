import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import { DrawerMenuListItem } from './DrawerMenuListItem';

describe('<DrawerMenuListItem />', () => {

    it('renders correctly', () => {
        const wrapper = shallow(<DrawerMenuListItem primaryContent="Text" />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('contains a secondary content div when there is secondary content', () => {
        const wrapper = shallow(
            <DrawerMenuListItem 
                primaryContent="Text"
                secondaryContent="abc"
            />
        );
        expect(wrapper.find('.DrawerMenuListItem__secondary-content').length).toBe(1);
    });

    it('does not contain a secondary content div when there is no secondary content', () => {
        const wrapper = shallow(<DrawerMenuListItem primaryContent="Text" />);
        expect(wrapper.find('.DrawerMenuListItem__secondary-content').length).toBe(0);
    });

});