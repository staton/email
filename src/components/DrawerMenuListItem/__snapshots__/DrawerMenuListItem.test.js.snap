// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerMenuListItem /> renders correctly 1`] = `
<li
  className="DrawerMenuListItem"
  onClick={[Function]}
  title=""
>
  <Link
    replace={false}
    to="/"
  >
    <div
      className="DrawerMenuListItem__container"
    >
      <div
        className="DrawerMenuListItem__left-container"
      >
        <div
          className="DrawerMenuListItem__icon"
        />
        <div
          className="DrawerMenuListItem__content"
        >
          <div
            className="DrawerMenuListItem__primary-content"
          >
            Text
          </div>
        </div>
      </div>
    </div>
  </Link>
</li>
`;
